core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -266,6 +266,10 @@ private[spark] object Logging {
       }
     }
 
+    private def loggerWithoutConfig(logger: Log4jLogger): Boolean = {
```

Comment:
Actually @viirya, I was thinking about similar approach initially but thought that this will break https://github.com/apache/spark/commit/5f11e8c4cb9a5db037ac239b8fcc97f3a746e772 because the appenders are added into the root logger if I am not mistaken.

Reply:
Hm? Not sure if I understand your concern. This check doesn't rely on the configuration of root logger. It only checks if the logger uses same LogConfig as root logger. Based on log4j2 architecture, if the logger has custom config, it will have a different LogConfig.

Reply:
For some reasons, `logger.getLevel` was `null`, and `logger.getAppenders` was empty in Log4J by default. I guess that it was only explicitly set when specified assuming from https://github.com/apache/spark/commit/61ebc8118665531b4c11a31f2b3d459dd201b097 (?). cc @sarutak.

After Log4J upgrade, `logger.getLevel` is set, for example, `INFO` and `logger.getAppenders` contains `STDOUT` presumably set from https://github.com/apache/spark/blob/ca5cabe5fbffdf02b51da2fd576ae8dc6d5b5909/core/src/main/resources/org/apache/spark/log4j2-defaults.properties#L23 (?).

I was wondering if we do something like:

```diff
- !logger.getAppenders.isEmpty
+ logger.getAppenders.asScala.find(_._1 != "STDOUT").isDefined
```

But not sure if this would catch all the details. I haven't closely followed up Log4J migration fixes so I might be wrong somewhere.

Reply:
Okie, maybe I should learn more :-). I am fine as long as it works. Would be great to have @sarutak's look here ..

Reply:
This is what I got after reading their architecture doc: https://logging.apache.org/log4j/2.x/manual/architecture.html, especially the section of `LoggerConfig`.

Reply:
I hope I don't misunderstand it. :)

Reply:
@HyukjinKwon Thank you for pinging me. I checked this part and I think the behavior seems not to be changed.

Reply:
Thanks for confirmation!